### ISSUE TITLE
Circular Reference Compilation Error when Compiling with Xcode 6.0 Beta Tools

### ISSUE THREAD

**Issue by mikelikespie** [gonzalezreal/swift-markdown-ui#323]
Thank you for contributing to MarkdownUI!

Before you submit an issue, please complete the report with all the relevant details for your bug and go through every item in the checklist.

**Describe the bug**

**Checklist**
- [X] I can reproduce this issue with a vanilla SwiftUI project.
- [X] I can reproduce this issue using the `main` branch of this package.
- [X] This bug hasn't been addressed in an [existing GitHub issue](https://github.com/gonzalezreal/swift-markdown-ui/issues).

**Steps to reproduce**
1. Using the new xcode 6 beta, try building the project or run `swift build` with new CLI tools selected

**Expected behavior**
It to build

**Screenshots**

```
error: emit-module command failed with exit code 1 (use -v to see invocation)
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
/Users/ml/code/swift-markdown-ui/Sources/MarkdownUI/Parser/InlineNode.swift:3:7: note: through reference here
 1 | import Foundation
 2 | 
 3 |  enum InlineNode: Hashable {
   |       |- note: through reference here
   |       |- note: through reference here
   |       `- note: through reference here
 4 |   case text(String)
 5 |   case softBreak
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
/Users/ml/code/swift-markdown-ui/Sources/MarkdownUI/Parser/InlineNode.swift:3:7: note: through reference here
 1 | import Foundation
 2 | 
 3 |  enum InlineNode: Hashable {
   |       |- note: through reference here
   |       |- note: through reference here
   |       `- note: through reference here
 4 |   case text(String)
 5 |   case softBreak
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: error: circular reference
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
<unknown>:0: note: through reference here
/Users/ml/code/swift-markdown-ui/Sources/MarkdownUI/Parser/InlineNode.swift:3:7: note: through reference here
 1 | import Foundation
 2 | 
 3 |  enum InlineNode: Hashable {
   |       |- note: through reference here
   |       |- note: through reference here
   |       `- note: through reference here
 4 |   case text(String)
 5 |   case softBreak
```

**Version information**
- MarkdownUI: c0daf6eb79d97964180f3113868c990bd1c4a007
- OS: macos 14.5 (23F79)
- Xcode: Version 16.0 beta (16A5171c)

**Additional context**

**Comment by electricsidecar-dev** [comment]
I've submitted FB13830058 to Apple for this bug.

I've isolated the bug down to the following minimally reproducible sample:

```swift
indirect enum BlockNode: Hashable {
  case paragraph(content: [InlineNode])
}

indirect enum InlineNode: Hashable {
  case emphasis(children: [InlineNode])
}
```

**Comment by electricsidecar-dev** [comment]
Oddly, the following snippet builds:

```
indirect enum BlockNode {
  case paragraph(content: InlineNode)
}

indirect enum InlineNode {
  case foo
  case emphasis(children: [InlineNode])
}
```

but this one does not:

```
indirect enum BlockNode {
  case paragraph(content: [InlineNode])
}

indirect enum InlineNode {
  case foo
  case emphasis(children: [InlineNode])
}
```

**Comment by electricsidecar-dev** [comment]
More fun: the following builds

```swift
indirect enum BlockNode {
  case paragraph([InlineNode])
}

indirect enum InlineNode {
  case foo
  case emphasis(children: [InlineNode])
}
```

but this does not:

```swift
indirect enum BlockNode {
  case paragraph(content: [InlineNode])
}

indirect enum InlineNode {
  case foo
  case emphasis(children: [InlineNode])
}
```

**Comment by electricsidecar-dev** [comment]
I've got a buildable patch working locally. Will send up a PR.

**Comment by JoshuaRileyDev** [comment]
How can I reference that PR in Xcode? Or does it need to be approved

**Comment by electricsidecar-dev** [comment]
Something like this in your Package.swift should work until this PR is merged:

```swift
    dependencies: [
        .package(url: "https://github.com/electricsidecar-dev/swift-markdown-ui.git", .branch("recursive")),
    ],
```

**Comment by gonzalezreal** [comment]
@electricsidecar-dev Thank you for investigating this and coming up with a workaround. I hope they fix it in the upcoming Xcode 16 betas. Great work! ❤️

**Comment by alexsllater** [comment]
please push update

**Comment by gonzalezreal** [comment]
Fixed by #328
